lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs

```diff
@@ -151,3 +153,11 @@ class KnownAddresses s where
 -- Hardened derivation is used at this level.
 coinTypeAda :: Index 'Hardened 'CoinTypeK
 coinTypeAda = toEnum 0x80000717
+
+-- It is used for geting purpose for a given key.
+class GetPurpose (key :: Depth -> Type -> Type)  where
+    getPurpose :: Index 'Hardened 'PurposeK
+
+-- It is used for geting account public key for a given state.
+class GetAccount s (key :: Depth -> Type -> Type) | s -> key  where
```

Comment:
Why the functional dependency here :thinking: ?

Reply:
we have s ~ SomethingKey n key, and return key ~Account XPub, and I wanted to enforce that key is from s...